# This is a (another) basic Python quick-start guide

Let's start with how to print things to the screen.

In [2]:
# Typing an expression and then running it with Shift+Enter will show the result
12 + 34

46

In [3]:
# However, if you want to see the result of multiple expressions, you can use print()
print(12 + 34)
print(56 + 78)

46
134


In [4]:
# Assign the result of an expression to a variable
x = 12 + 34
print(x)

46


As you have already seen, you can use Python to perform basic math operations:

In [5]:
1+2

3

In [6]:
1-2

-1

In [7]:
1*2

2

In [8]:
1/2

0.5

Note that the syntax for exponents is NOT `a^b`, but rather `a**b`. `a^b` stands for Exclusive OR (XOR)

In [9]:
# 2 to the power of 4
powerResult = 2**4
# You can use the a^(1/b) to get the bth root of a
rootResult = 8**(1/3)
print(powerResult, rootResult)

16 2.0


Python also follows the order of operations.

In [10]:
# Complex order of operations example
print(2 + 3 * (4 - 5) ** 6 / (1/3) * 8 + 9)

83.0


## Math Module

You can get more access to Math functions and constants by using the `math` module:

In [3]:
# Below line imports the math module
import math

### Math Constants

In [ ]:
# Mathematical constants
print(math.pi)
print(math.sin(math.pi/2))
print(math.cos(math.pi/2))
print(math.log(100, 10))
# You can also use math.sqrt() as an alternative to the ** operator
print(math.sqrt(100))

## Common Basic Math Functions

### Absolute Value

In [6]:
# Absolute value
print(abs(-5))

5


### Rounding

In [9]:
# pi as an example variable for the following examples
pi = math.pi
print(pi)

3.141592653589793


In [12]:
# Round up to next integer
print("Ceil", math.ceil(pi))
# Round down to previous integer
print("Floor", math.floor(pi))

# Round to nearest integer
print("Round", round(pi))

# Round to nearest integer with specified number of decimal places
print("Round (2 dec)", round(pi, 2)) # 2 decimal places
print("Round (3 dec)", round(pi, 3)) # 3 decimal places

Ceil 4
Floor 3
Round 3
Round (2 dec) 3.14
Round (3 dec) 3.142


### Trigonometry and Logarithms

In [14]:
# sin
print("sin", math.sin(pi/2))
# cos
print("cos", math.cos(pi/2))
# tan
print("tan", math.tan(pi/4))

# asin
print("asin", math.asin(1))
# acos
print("acos", math.acos(0))
# atan
print("atan", math.atan(1))

# log
print("log", math.log(100, 10))
# log10
print("log10", math.log10(100))
# log2
print("log2", math.log2(100))

sin 1.0
cos 6.123233995736766e-17
tan 0.9999999999999999
asin 1.5707963267948966
acos 1.5707963267948966
atan 0.7853981633974483
log 2.0
log10 2.0
log2 6.643856189774724


#### Did you notice something odd with the output of the above functions, particularly cos and tan?

- Starting with `tan`, the output of $\tan{\frac{\pi}{4}}$ should have been 1, but it was 0.99999999999999989. This is because Python uses floating point numbers (floats) to represent real numbers. Floating point numbers are not exact, and so they can't represent all real numbers exactly. This is why you should be careful when comparing floating point numbers for equality. You can work around this by rounding the numbers to a certain number of decimal places (and in practical situations involving real/ observed data, you should be doing this anyway because of SigFigs).

- Similarly, with `cos`, the output of $\cos{\frac{\pi}{2}}$ should have been 0, but it was $6.123\times10^{-16}$ (represented with notation `6.123e-17`). This is because the computer can't represent $\frac{\pi}{2}$ exactly, so it has to approximate it. This approximation is not exact, and so the result is not exactly 0. Once again, you can work around this by rounding the numbers to a certain number of decimal places.

In [17]:
# Reprinted incorrect example from above:
print("cos", math.tan(pi/4))
print("sin", math.cos(pi/2))

# Corrected example by rounding to 3 decimal places
print("cos", round(math.tan(pi/4), 3))
print("sin", round(math.cos(pi/2), 3))

cos 0.9999999999999999
sin 6.123233995736766e-17
cos 1.0
sin 0.0


### More Math Module Things (Full Documentation)
Link: [Python `math` Module Overview/ Interactive Documentation](https://www.w3schools.com/python/module_math.asp)<br>Click the method/ mathematical constant names to see the documentation and examples for each command in the `math` module.

## Congrats! You now know how to use Python to turn your computer work into an overpriced calculator!